File: Data Structures & Algorithms/two-integer-sum-ii/submission-6.cs
public class Solution {
    public int[] TwoSum(int[] numbers, int target) {
        int[] results = new int[2];
        int l = 0;
        int r = numbers.Length - 1;

        while(l <= r)
        {
            int currentTarget = numbers[l] + numbers[r];
            if(currentTarget < target)
            {
                l++;
            }
            else if(currentTarget > target)
            {
                r--;
            }
            else
            {
                results[0] = l + 1;
                results[1] = r + 1;
                return results;
            }
        }
        return results;
            }
}

File: Data Structures & Algorithms/two-integer-sum/submission-6.py
class Solution:
    def twoSum(self, nums, target):
        _dict = {}
        indexes = []

        for i,n in enumerate(nums):
            target_num = target - n
            if target_num in _dict:
                indexes.append(_dict.get(target_num))
                indexes.append(i)
            else:
                _dict[n] = i
        return indexes

